iroh-relay/src/server/clients.rs

```diff
@@ -1,228 +1,129 @@
             old_client.shutdown().await;
         }
     }
 
     /// Removes the client from the map of clients, & sends a notification
     /// to each client that peers has sent data to, to let them know that
     /// peer is gone from the network.
-    pub async fn unregister(&mut self, peer: &NodeId) {
-        trace!("unregistering client: {:?}", peer);
-        if let Some(client) = self.inner.remove(peer) {
-            for key in client.sent_to.iter() {
-                self.send_peer_gone(key, *peer);
+    async fn unregister(&self, node_id: NodeId) {
+        trace!(node_id = node_id.fmt_short(), "unregistering client");
+
+        if let Some((_, client)) = self.0.clients.remove(&node_id) {
+            if let Some((_, sent_to)) = self.0.sent_to.remove(&node_id) {
+                for key in sent_to {
+                    match client.send_peer_gone(key) {
+                        Ok(_) => {}
+                        Err(SendError::PacketDropped) => {
+                            warn!("client {key:?} too busy to receive packet, dropping packet");
+                        }
+                        Err(SendError::SenderClosed) => {
+                            debug!("can no longer write to client {key:?}, dropping packet");
+                        }
+                    }
+                }
             }
-            warn!("pruning connection {peer:?}");
             client.shutdown().await;
         }
     }
 
-    /// Attempt to send a packet to client with [`NodeId`] `key`
-    pub async fn send_packet(&mut self, key: &NodeId, packet: Packet) -> Result<()> {
-        if let Some(client) = self.inner.get(key) {
-            let res = client.send_packet(packet);
-            return self.process_result(key, res).await;
-        }
-        bail!("Could not find client for {key:?}, dropped packet");
-    }
-
-    pub async fn send_disco_packet(&mut self, key: &NodeId, packet: Packet) -> Result<()> {
-        if let Some(client) = self.inner.get(key) {
-            let res = client.send_disco_packet(packet);
-            return self.process_result(key, res).await;
-        }
-        bail!("Could not find client for {key:?}, dropped packet");
-    }
-
-    fn send_peer_gone(&mut self, key: &NodeId, peer: NodeId) {
-        if let Some(client) = self.inner.get(key) {
-            let res = client.send_peer_gone(peer);
-            let _ = self.process_result_no_fallback(key, res);
-            return;
+    /// Attempt to send a packet to client with [`NodeId`] `dst`
+    pub(super) async fn send_packet(&self, dst: NodeId, data: Bytes, src: NodeId) -> Result<()> {
+        if let Some(client) = self.0.clients.get(&dst) {
+            let res = client.send_packet(src, data);
+            return self.process_result(src, dst, res).await;
         }
-        warn!("Could not find client for {key:?}, dropping peer gone packet");
+        warn!("could not find client for {dst:?}, dropped packet");
```

Comment:
```suggestion
        debug!(dst = dst.fmt_short(), "No connected client, dropped packet");
```
This is not an error of the server, just floods the server logs.  Possibly even debug is too high if this happens at high volume, but that's an ops thing that can be sorted out later.  I just couldn't skip this one as we'd have to reduce the log level of this anyway at some point.

Reply:
done